chia/rpc/wallet_rpc_api.py

```diff
@@ -1289,91 +1306,56 @@ async def did_transfer_did(self, request):
     # NFT Wallet
     ##########################################################################################
 
-    async def nft_mint_nft(self, request):
-        wallet_id = int(request["wallet_id"])
+    async def nft_mint_nft(self, request) -> Dict:
+        log.debug("Got minting RPC request: %s", request)
+        wallet_id = uint32(request["wallet_id"])
+        assert self.service.wallet_state_manager
         nft_wallet: NFTWallet = self.service.wallet_state_manager.wallets[wallet_id]
-        address = request["artist_address"]
+        address = request.get("artist_address")
         if isinstance(address, str):
-            address = decode_puzzle_hash(address)
+            puzzle_hash = decode_puzzle_hash(address)
+        elif address is None:
+            puzzle_hash = await nft_wallet.standard_wallet.get_new_puzzlehash()
+        else:
+            puzzle_hash = address
         metadata = Program.to(
             [
                 ("u", request["uris"]),
                 ("h", request["hash"]),
             ]
         )
+        nft_record = await nft_wallet.generate_new_nft(metadata, puzzle_hash)
+        return {"wallet_id": wallet_id, "success": True, "nft": nft_record}
 
-        await nft_wallet.generate_new_nft(metadata, request["artist_percentage"], address)
-        return {"wallet_id": wallet_id, "success": True}
-
-    async def nft_get_nfts(self, request):
-        wallet_id = int(request["wallet_id"])
+    async def nft_get_nfts(self, request) -> Dict:
+        wallet_id = uint32(request["wallet_id"])
+        assert self.service.wallet_state_manager
         nft_wallet: NFTWallet = self.service.wallet_state_manager.wallets[wallet_id]
         nfts = nft_wallet.get_current_nfts()
-        nft_info_list = []
+        nft_uri_pairs = []
         for nft in nfts:
-            nft_info_list.append(get_nft_info_from_puzzle(nft.full_puzzle, nft.coin))
-        return {"wallet_id": wallet_id, "success": True, "nft_list": nft_info_list}
+            uri = get_uri_list_from_puzzle(nft.full_puzzle)
```

Comment:
This will break the GUI integration, as the frontend is expecting an array of NFTInfo objects, but we can fix it in a follow-up PR.